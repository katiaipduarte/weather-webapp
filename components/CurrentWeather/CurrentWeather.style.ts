import styled from 'styled-components'

export const WeatherInfoContainer = styled.section`
	display: flex;
	align-items: end;
	line-height: 5rem;
	flex-flow: row;
	color: white;
	margin-top: 50vh;
	margin: 55vh 0 5vh 0;

	h2 {
		margin-right: 1.5rem;
		font-weight: 500;
	}

	.location-column {
		margin-right: 1.5rem;
		line-height: 1.75rem;

		h3 {
			letter-spacing: -0.1rem;
			font-weight: 400;
		}
	}

	.weather-column {
		line-height: 23px;

		.icon-image {
			margin-bottom: -10px !important;
		}
	}
`
