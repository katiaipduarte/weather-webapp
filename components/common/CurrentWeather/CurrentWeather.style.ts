import styled from 'styled-components'

export const WeatherInfoContainer = styled.section`
	display: flex;
	align-items: end;
	line-height: 5rem;
	flex-flow: row;
	color: white;
	margin: 5rem auto 0;

	h2 {
		margin-right: 1.5rem;
		font-weight: 500;
	}

	.location-current-weather {
		position: relative;
		margin-right: 1.75rem;

		.location-column {
			line-height: 1.75rem;

			h3 {
				letter-spacing: -0.1rem;
				font-weight: 400;
			}
		}
	}

	.weather-column {
		line-height: 23px;

		.icon-image {
			margin-bottom: -10px !important;
		}

		p {
			margin-bottom: 4px;
		}
	}

	@media (min-width: 960px) {
		margin: 0 2.5rem !important;
	}
`
