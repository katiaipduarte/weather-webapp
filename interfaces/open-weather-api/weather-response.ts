import { CurrentWeather } from './current-weather'
import { DailyWeather } from './daily-weather'

export interface WeatherResponse {
	lat: number
	lon: number
	timezone: string
	timezone_offset: number
	current: CurrentWeather
	daily: DailyWeather[]
}
