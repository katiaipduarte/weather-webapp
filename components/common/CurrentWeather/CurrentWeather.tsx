import { CurrentWeather } from '@interfaces/open-weather-api/current-weather'
import { GPSLocation } from '@interfaces/open-weather-api/location'
import { GlobalState } from '@store/store'
import Image from 'next/image'
import { useSelector } from 'react-redux'
import FavouriteButton from '../FavouriteButton/FavouriteButton'
import { WeatherInfoContainer } from './CurrentWeather.style'

type Props = {
	currentWeather: CurrentWeather
	location: GPSLocation
}

const CurrentWeather = (props: Props) => {
	const { currentWeather, location } = props
	const favourites = useSelector((state: GlobalState) => state.favouritesState)
	const isFavourite = favourites.favourites.find(
		(i: GPSLocation) => i.lat === location.lat && i.lon === location.lon
	)

	const symbol = '\u00b0'

	const getIconSrc = (): string => {
		return `http://openweathermap.org/img/w/${currentWeather.weather[0].icon}.png`
	}

	const getDate = (): string => {
		const options = {
			weekday: 'long',
			year: '2-digit',
			month: 'long',
			day: 'numeric',
		} as Intl.DateTimeFormatOptions

		const date = new Date()

		return date.toLocaleDateString(undefined, options)
	}

	return (
		<WeatherInfoContainer>
			<h2>
				{Math.round(currentWeather.temp)}
				{symbol}
			</h2>
			<div className='location-column'>
				<h3>{location.name}</h3>
				<p>{getDate()}</p>
			</div>
			<div className='weather-column'>
				<Image
					src={getIconSrc()}
					width='40px'
					height='40px'
					className='icon-image'
					alt={currentWeather.weather[0].main}
				/>
				<p>{currentWeather.weather[0].main}</p>
			</div>
			<FavouriteButton location={location} status={!!isFavourite} />
		</WeatherInfoContainer>
	)
}

export default CurrentWeather
