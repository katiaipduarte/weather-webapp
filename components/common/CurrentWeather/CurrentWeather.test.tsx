import { mockLocationResponse } from '@mocks/location-response.mock'
import { mockWeatherResponse } from '@mocks/weather-response.mock'
import { render } from '@testing-library/react'
import CurrentWeather from './CurrentWeather'

describe('<CurrentWeather /> component Unit Test', () => {
	it('should render 1 <CurrentWeather /> component', () => {
		const { container } = render(
			<CurrentWeather
				currentWeather={mockWeatherResponse.current}
				location={mockLocationResponse}
			/>
		)
		expect(container).toMatchSnapshot()
	})
})
