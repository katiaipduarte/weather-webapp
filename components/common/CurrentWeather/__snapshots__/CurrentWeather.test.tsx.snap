// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrentWeather /> component Unit Test should render 1 <CurrentWeather /> component 1`] = `
<div>
  <section
    class="sc-gsnTZi fdzpQT"
  >
    <h2>
      18
      Â°
    </h2>
    <div
      class="location-current-weather"
    >
      <div
        class="location-column"
      >
        <h3>
          Leiria
        </h3>
        <p>
          Friday, April 22, 22
        </p>
      </div>
    </div>
    <div
      class="weather-column"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2740%27%20height=%2740%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Clouds"
          class="icon-image"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <p>
        Clouds
      </p>
    </div>
  </section>
</div>
`;
